core/primitives/src/validator_mandates/compute_price.rs

```diff
@@ -0,0 +1,246 @@
+use {
+    super::ValidatorMandatesConfig,
+    near_primitives_core::types::Balance,
+    std::cmp::{min, Ordering},
+};
+
+/// Given the stakes for the validators and the target number of mandates to have,
+/// this function computes the mandate price to use. It works by iterating a
+/// function in an attempt to find its fixed point. This function is motived as follows:
+/// Let the validator stakes be denoted by `s_i` a let `S = \sum_i s_i` be the total
+/// stake. For a given mandate price `m` we can write each `s_i = m * q_i + r_i`
+/// (by the Euclidean algorithm). Hence, the number of whole mandates created by
+/// that price is equal to `\sum_i q_i`. If we set this number of whole mandates
+/// equal to the target number `N` then substitute back in to the previous equations
+/// we have `S = m * N + \sum_i r_i`. We can rearrange this to solve for `m`,
+/// `m = (S - \sum_i r_i) / N`. Note that `r_i = a_i % m` so `m` is not truly
+/// isolated, but rather the RHS is the expression we want to find the fixed point for.
+pub fn compute_mandate_price<F, I>(config: ValidatorMandatesConfig, stakes: F) -> Balance
```

Comment:
I think `stakes` can just be a `Vec` which will be cloned every time needed. Function makes signature more complex and I don't see where container wouldn't work.

Reply:
Done in https://github.com/near/nearcore/pull/11044/commits/1f2f35c4af21bb0dad75f3c56e32476f1f6ff561